src/App.js

```diff
@@ -1,23 +1,148 @@
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-          <SentenceContainer/>
-      </header>
-        <div>
-            TODO:
-            <li>- SHow guess value in setence with green or red correct/incorrect</li>
-                <li>- settings/config</li>
-        </div>
-        <SettingsContainer/>
-    </div>
-  );
+class App extends Component
+{
+
+    getAllKeysWithTrueValue = (obj) => {
+        let out = [];
+        let keys = Object.keys(obj);
+        for (let i of keys) {
+            if (obj[i] === true){
+                out.push(i);
+            }
+        }
+        return out;
+    }
+
+    componentWillMount() {
+        this.resetCurrentKeysByFilters();
+    }
+
+    resetCurrentKeysByFilters() {
+        let filterScalesByAccidental = (scaleContainer, allowedAccidentals = ['sharp', 'natural', 'flat']) => {
+            let filteredScales = [];
+            // TODO: Refactor this to a single if with all conditions in a single test
+            // TODO: Refactor this to use accidentals object instead of string matching
+            scaleContainer.forEach(function (scale) {
+                if ((scale[0].includes('#')) && isInArray('sharp', allowedAccidentals)) {
+                    filteredScales.push(scale);
+                }
+                else if ((scale[0].includes('b')) && isInArray('flat', allowedAccidentals)) {
+                    filteredScales.push(scale);
+                }
+                else if (((!scale[0].includes('#')) && (!scale[0].includes('b'))) && isInArray('natural', allowedAccidentals)) {
+                    filteredScales.push(scale);
+                }
+            });
+            return filteredScales;
+        }
+
+        let filteredMajorScales = filterScalesByAccidental(allScales.major, this.getAllKeysWithTrueValue(this.state.keyAccidentals));
+        let filteredMinorScales = filterScalesByAccidental(allScales.minor, this.getAllKeysWithTrueValue(this.state.keyAccidentals));
+
+        let allFilteredScales = {
+            major: filteredMajorScales,
+            minor: filteredMinorScales,
+        };
+
+        let buildFilteredKeys = (keyQuality) => {
+            let allKeys = {};
+            for (let scale of allFilteredScales[keyQuality]) {
+                // Build key
+                let chords = [];
+                let keyName = ((keyQuality === 'major') ? scale[0] : scale[0] + "m");
+                for (let noteString of scale) {
+                    //Build chord
+                    let noteInterval = scale.indexOf(noteString);
+                    let noteQuality = keyFormulas[keyQuality][noteInterval];
+                    let chord = {
+                        nashvilleRoman: nashvilleNumbers.roman[noteInterval],
+                        nashvilleArabic: nashvilleNumbers.arabic[noteInterval],
+                        chordFullName: noteString + " " + noteQuality.fullName,
```

Comment:
Good opportunity to do some ES6 string interpolation:

```suggestion
                        chordFullName: `${noteString} ${noteQuality.fullName}`,
```